src/opentype.mjs

```diff
@@ -310,8 +310,13 @@ function parseBuffer(buffer, opt={}) {
                 metaTableEntry = tableEntry;
                 break;
             case 'gasp':
-                table = uncompressTable(data, tableEntry);
-                font.tables.gasp = gasp.parse(table.data, table.offset);
+                try {
```

Comment:
not sure if the try/catch is still needed here
and if it is needed, maybe we shall have it for every cases to keep a unified behavior

Reply:
I believe that increasing the scope of OpenType.js to include a bunch of optional tables necessitates making them fail gracefully if they cannot be parsed.  The primary issue in my case is not that `gasp` could not be parsed correctly (I don't care about this table at all), but rather that my program crashed simply because this optional table could not be parsed.  Even after the fixes in this PR, exceptions during parsing this table may occur due to (1) undiscovered bugs in the parsing/writing functions or (2) flaws in the `gasp` table of otherwise valid user-provided fonts.  FontForge handles failures when parsing optional tables by skipping and printing a warning message, which I think makes sense. 

If you want, I can update such that all optional tables are handled in a similar way.  I only added for `gasp` as I see that table as higher-risk (as it was added the most recently without being fully tested), but I don't think crashing when an optional table cannot be parsed is ever a desirable behavior.

Reply:
I would have expected the inner parse function to do the console.war/error + return null
But I just saw that `gasp.parse` had a `check.argument` that could `throw`
so my comment is irrelevant and the try/catch is well used here.
And I agree with you that we shall be error-resilient.

Reply:
> I believe that increasing the scope of OpenType.js to include a bunch of optional tables necessitates making them fail gracefully if they cannot be parsed.

Graceful table handling will be implemented via #704